fuel-vm/src/checked_transaction.rs

```diff
@@ -186,40 +191,91 @@ pub trait IntoChecked: FormatValidityChecks + Sized {
+            max_gas_per_predicate: consensus_params
+                .predicate_params()
+                .max_gas_per_predicate,
+            max_gas_per_tx: consensus_params.tx_params().max_gas_per_tx,
+            max_inputs: consensus_params.tx_params().max_inputs,
+            contract_max_size: consensus_params.contract_params().contract_max_size,
+            max_message_data_length: consensus_params
+                .predicate_params()
+                .max_message_data_length,
+            tx_offset: consensus_params.tx_params().tx_offset(),
+            fee_params: *consensus_params.fee_params(),
+        };
+        self.into_checked_basic(block_height, consensus_params, &chain_id)?
+            .check_signatures(&chain_id)?
+            .check_predicates(check_predicate_params)
     }
 
     /// Returns transaction that passed only `Checks::Basic`.
     fn into_checked_basic(
         self,
         block_height: BlockHeight,
-        params: &ConsensusParameters,
+        consensus_params: ConsensusParams<'_>,
+        chain_id: &ChainId,
     ) -> Result<Checked<Self>, CheckError>;
 }
 
+/// The parameters needed for checking a predicate
+#[derive(Debug, Clone)]
+pub struct CheckPredicateParams {
+    /// Gas costs for opcodes
+    pub gas_costs: GasCosts,
+    /// Chain ID
+    pub chain_id: ChainId,
+    /// Maximum gas per predicate
+    pub max_gas_per_predicate: u64,
+    /// Maximum gas per transaction
+    pub max_gas_per_tx: u64,
+    /// Maximum number of inputs
+    pub max_inputs: u64,
+    /// Maximum size of the contract in bytes
+    pub contract_max_size: u64,
+    /// Maximum length of the message data
+    pub max_message_data_length: u64,
+    /// Offset of the transaction data in the memory
+    pub tx_offset: usize,
+    /// Fee parameters
+    pub fee_params: FeeParameters,
+}
+
+impl Default for CheckPredicateParams {
+    fn default() -> Self {
+        CheckPredicateParams {
+            gas_costs: Default::default(),
+            chain_id: Default::default(),
+            max_gas_per_predicate: PredicateParameters::DEFAULT.max_gas_per_predicate,
+            max_gas_per_tx: TxParameters::DEFAULT.max_gas_per_tx,
+            max_inputs: TxParameters::DEFAULT.max_inputs,
+            contract_max_size: ContractParameters::DEFAULT.contract_max_size,
+            max_message_data_length: PredicateParameters::DEFAULT.max_message_data_length,
+            tx_offset: TxParameters::DEFAULT.tx_offset(),
+            fee_params: Default::default(),
+        }
+    }
+}
+
 /// Provides predicate verification functionality for the transaction.
 #[async_trait::async_trait]
 pub trait CheckPredicates: Sized {
     /// Performs predicates verification of the transaction.
-    fn check_predicates(
-        self,
-        params: &ConsensusParameters,
-        gas_costs: &GasCosts,
-    ) -> Result<Self, CheckError>;
+    fn check_predicates(self, params: CheckPredicateParams) -> Result<Self, CheckError>;
```

Comment:
I prefer `&CheckPredicateParams` more than `CheckPredicateParams` because we usually store this configuration somewhere. Plus, downstream code decides whether it needs to clone or not.